training/src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -216,6 +223,41 @@ void whenFindByPublisherAndGenreAndYearAndPassParams_thenReturnABook() throws Ex
             .andExpect(jsonPath("$[0].title", is(oneTestBook.getTitle())));
     }
 
+    @WithMockUser(value = "pedro")
+    @Test
+    void whengetAllByConditionsWithPageable_thenReturnsBooks() throws Exception {
+
+        Pageable pageable = PageRequest.of(0, 2, Sort.by(
+            Order.asc("id")));
+        Mockito.when(mockBookRepository.getAllByConditions(
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            pageable
+        )).thenReturn(java.util.Optional.ofNullable(oneArrayTest));
+
+        String url = ("/api/books");
+
+        mvc.perform(get(url)
+            .contentType(MediaType.APPLICATION_JSON)
+            .param("page", "0")
+            .param("size", "2")
+            .param("sort", "id,asc")
+        )
+            .andExpect(status().isOk())
```

Comment:
You could also check that the `$` size is 2 just to be absolutely sure that the size param is working.